src/ibm/hypervisor-network-mgr-src/hyp_ethernet_interface.cpp

```diff
@@ -21,6 +21,138 @@ constexpr char biosIntType[] =
 constexpr char biosEnumType[] =
     "xyz.openbmc_project.BIOSConfig.Manager.AttributeType.Enumeration";
 
+std::string HypEthInterface::getAttrFromBiosTable(std::string attrName)
+{
+    constexpr auto BIOS_SERVICE = "xyz.openbmc_project.BIOSConfigManager";
+    constexpr auto BIOS_OBJPATH = "/xyz/openbmc_project/bios_config/manager";
+    constexpr auto BIOS_MGR_INTF = "xyz.openbmc_project.BIOSConfig.Manager";
+
+    try
+    {
+        using getAttrRetType =
+            std::tuple<std::string, std::variant<std::string, int64_t>,
+                       std::variant<std::string, int64_t>>;
+        getAttrRetType ip;
+        auto method = bus.new_method_call(BIOS_SERVICE, BIOS_OBJPATH,
+                                          BIOS_MGR_INTF, "GetAttribute");
+
+        method.append(attrName);
+
+        auto reply = bus.call(method);
+
+        std::string type;
+        std::variant<std::string, int64_t> currValue;
+        std::variant<std::string, int64_t> defValue;
+        reply.read(type, currValue, defValue);
+        return std::get<std::string>(currValue);
+    }
+    catch (const sdbusplus::exception::SdBusError& ex)
+    {
+        log<level::ERR>("Failed to get the hostname from bios table",
+                        entry("ERR=%s", ex.what()));
+    }
+    return "";
+}
+
+void HypEthInterface::watchBaseBiosTable()
+{
+    auto BIOSAttrUpdate = [&](sdbusplus::message::message& m) {
+        std::map<std::string, std::variant<BiosBaseTableType>>
+            interfacesProperties;
+
+        std::string objName;
+        m.read(objName, interfacesProperties);
+
+        // Check if the property change signal is for BaseBIOSTable property
+        // If found, proceed; else, continue to listen
+        auto find = interfacesProperties.find("BaseBIOSTable");
+        if (find == interfacesProperties.end())
+        {
+            // Return & continue to listen
+            return;
+        }
+        log<level::INFO>("BaseBIOSTable - property changed");
+
+        // Check if the IP address has changed (i.e., if current ip address in
+        // the biosTableAttrs data member and ip address in bios table are
+        // different)
+        auto biosTableAttrs = manager.getBIOSTableAttrs();
+        for (auto i : biosTableAttrs)
+        {
+            if (i.first == "vmi_if0_ipv4_ipaddr" ||
+                i.first == "vmi_if1_ipv4_ipaddr")
+            {
+                std::string currIpAddr = std::get<std::string>(i.second);
+                if (currIpAddr.empty())
+                {
+                    log<level::INFO>("Current IP in biosAttrs copy is empty");
+                    return;
+                }
+
+                std::string ipAddr = getAttrFromBiosTable(i.first);
```

Comment:
I am confused about how this worked.

you are checking the value coming in property changed signal of base bios table  - to the value which is already present in the biostable.

Both of them will be the same right ?

you should be checking the current value on your hypervisor dbus object to what is coming in the signal ?

Reply:
No. What I am checking is:
ipAddr - current value from the bios table
currIpAddr - there is a local copy of the bios table attrs and its values in hypEthernetInterface class (that stores the current dbus obj values). I am getting that ip address as currIpAddr and checking it against ipAddr

For example: (as mentioned in the commit message)
10.5.5.10 is the older ip and "currIpAddr" would be 10.5.5.10
ipAddr would be 10.5.5.23 (which is the new updated ip)